src/Ledger/Utxo/Properties.lagda

```diff
@@ -544,3 +548,194 @@ pov {tx}{utxo}{_}{fees}{deposits}{donations} h'
+    getCoin (deps ∪⁺ ❴ (dp , c) ❵ᵐ)       ≡⟨ gc-hom deps ❴ (dp , c) ❵ᵐ ⟩
+    getCoin deps + getCoin ❴ (dp , c) ❵ᵐ  ≡⟨ cong (getCoin deps +_) (getCoin-singleton (dp , c)) ⟩
+    getCoin deps + c                      ∎
+    where open Prelude.≡-Reasoning
+
+  ≤updatePropDeps : (props : List GovProposal) → ∀ {deposits} {txid} {gaDep}
+    → getCoin deposits ≤ getCoin (updateProposalDeposits props txid gaDep deposits)
+  ≤updatePropDeps [] = ≤-reflexive refl
+  ≤updatePropDeps (x ∷ props) = ≤-trans (≤updatePropDeps props)
+                                        (≤-trans (m≤m+n _ _)
+                                                 (≤-reflexive $ sym $ ∪⁺singleton≡))
+
+  updatePropDeps≡ : (ps : List GovProposal)(ds : DepositPurpose ⇀ Coin)
+    → ∀ {txi} {gaD}
+    → getCoin (updateProposalDeposits ps txi gaD ds) - getCoin ds ≡ (length ps) * gaD
+  updatePropDeps≡ [] ds = n∸n≡0 (getCoin ds)
+  updatePropDeps≡ (_ ∷ ps) ds {txi} {gaD} = let
+    upD = updateProposalDeposits ps txi gaD ds in
+    begin
+      getCoin (upD ∪⁺ ❴ GovActionDeposit (txi , length ps) , gaD ❵ᵐ) ∸ getCoin ds
+        ≡⟨ cong (_∸ getCoin ds) ∪⁺singleton≡ ⟩
+      getCoin upD + gaD ∸ getCoin ds
+        ≡⟨ +-∸-comm _ (≤updatePropDeps ps) ⟩
+      (getCoin upD ∸ getCoin ds) + gaD
+        ≡⟨ cong (_+ gaD) (updatePropDeps≡ ps ds) ⟩
+      (length ps) * gaD + gaD
+        ≡⟨ +-comm ((length ps) * gaD) gaD ⟩
+      gaD + (length ps) * gaD  ∎
+      where open Prelude.≡-Reasoning
+
+  module _
+    -- ASSUMPTION --
+    {- 2 -} {⊆→∪⁺ : (d d' : DepositPurpose ⇀ Coin) → d ˢ ⊆ d' ˢ → getCoin d' ≡ (getCoin d + getCoin (d' ∣ (dom d) ᶜ))}
+    where
+    getCoin-⊆ : (d d' : DepositPurpose ⇀ Coin) → d ˢ ⊆ d' ˢ → getCoin d ≤ getCoin d'
+    getCoin-⊆ d d' h = begin
+      getCoin d                             ≤⟨ m≤m+n (getCoin d) (getCoin (d' ∣ (dom d) ᶜ)) ⟩
+      getCoin d + getCoin (d' ∣ (dom d) ᶜ)  ≡⟨ sym (⊆→∪⁺ d d' h) ⟩
+      getCoin d'                            ∎
+      where open ≤-Reasoning
+
+    ∣∅ˢ≡id : {d : DepositPurpose ⇀ Coin} → getCoin d ≡ getCoin (d ∣ ∅ˢ ᶜ)
+    ∣∅ˢ≡id {d} = ≤-antisym (getCoin-⊆ d (d ∣ ∅ˢ ᶜ) $ proj₁ (swap $ resᵐ-∅ᶜ {M = d}))
+                           (getCoin-⊆ (d ∣ ∅ˢ ᶜ) d $ proj₂ (swap $ resᵐ-∅ᶜ {M = d}))
+
+    ∪⁺∅ᵐ∣∅ˢid : {d : DepositPurpose ⇀ Coin} → getCoin d ≡ getCoin ((d ∪⁺ ∅ᵐ) ∣ ∅ˢ ᶜ)
+    ∪⁺∅ᵐ∣∅ˢid {d} = trans ∪⁺∅ᵐ≡id (∣∅ˢ≡id {d ∪⁺ ∅ᵐ})
+
+    ≤certDeps : (certs : List DCert) {d :  DepositPurpose ⇀ Coin} ((dp , c) : DepositPurpose × Coin)
+      →  getCoin d ≤ getCoin ((d ∪⁺ ❴ (dp , c) ❵ᵐ) ∣ ∅ˢ ᶜ )
+    ≤certDeps certs {d} (dp , c) = begin
+      getCoin d                              ≤⟨ m≤m+n (getCoin d) c ⟩
+      getCoin d + c                          ≡⟨ sym ∪⁺singleton≡ ⟩
+      getCoin (d ∪⁺ ❴ (dp , c) ❵ᵐ)           ≡⟨ ∣∅ˢ≡id  {d ∪⁺ ❴ (dp , c) ❵ᵐ} ⟩
+      getCoin ((d ∪⁺ ❴ (dp , c) ❵ᵐ) ∣ ∅ˢ ᶜ ) ∎
+      where open ≤-Reasoning
+
+    ≤updateCertDeps : (certs : List DCert) {pp : PParams} {deposits :  DepositPurpose ⇀ Coin}
```

Comment:
This proof is pretty noisy. Removing the unused arguments in each case or replacing them with `_` probably helps a bit.